File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingOverrideHistoryModel.java
package kore.botssdk.models;

public class BrandingOverrideHistoryModel {
    private Boolean enable;
    private BrandingOverrideHistoryRecentModel recent;
    private BrandingOverrideHistoryRecentModel paginated_scroll;

    public BrandingOverrideHistoryModel updateWith(BrandingOverrideHistoryModel configModel) {
        enable = configModel.enable != null ? configModel.enable : enable;
        recent = recent != null && configModel.recent != null ? recent.updateWith(configModel.recent) : recent;
        paginated_scroll = paginated_scroll != null && configModel.paginated_scroll != null ? paginated_scroll.updateWith(configModel.paginated_scroll) : paginated_scroll;
        return this;
    }

    public BrandingOverrideHistoryRecentModel getRecent() {
        return recent;
    }

    public boolean isEnable() {
        return enable != null ? enable : false;
    }

    public BrandingOverrideHistoryRecentModel getPaginated_scroll() {
        return paginated_scroll;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingOverrideHistoryModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingOverrideHistoryModel(
    @SerializedName("isEnable")
    val isEnable: Boolean? = false,
    @SerializedName("recent")
    val recent: BrandingOverrideHistoryRecentModel? = null,
    @SerializedName("paginated_scroll")
    val paginatedScroll: BrandingOverrideHistoryRecentModel? = null
)

fun BrandingOverrideHistoryModel.updateWith(configModel: BrandingOverrideHistoryModel?): BrandingOverrideHistoryModel {
    return this.copy(
        isEnable = configModel?.isEnable ?: isEnable,
        recent = recent?.updateWith(configModel?.recent) ?: recent,
        paginatedScroll = paginatedScroll?.updateWith(configModel?.paginatedScroll) ?: paginatedScroll
    )
}

